### PR TITLE
Port to jbuilder

### PR DESCRIPTION
**Pull request by mseri**
Includes restructuring stdext and the move from xstringext to astring. Does not require a new spec file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,3 @@
-*.native
-*.swp
-
-setup.bin
-setup.data
-setup.log
-
 _build/
+.merlin
+*.install
```
Comment (lindig) [inline .gitignore]: Missing `\n`

File: Makefile [hunk 2]
```diff
@@ -1,51 +1,32 @@
-.PHONY: all clean install build
-all: build doc
+OPAM_PREFIX?=$(DESTDIR)$(shell opam config var prefix)
+OPAM_LIBDIR?=$(DESTDIR)$(shell opam config var lib)
 
-J=4
+.PHONY: release build install uninstall clean test doc reindent
 
-export OCAMLRUNPARAM=b
+release:
```
Comment (lindig) [inline Makefile]: Not sure about `release` as the default target. During development I am most likely to use `make` and not `make build` and thus I will miss out on the stricter checks. Whereas in the spec file we could to a `make release`.

Comment (mseri) [reply Makefile]: If I use release, we do not need to change the spec file at all

Comment (lindig) [reply Makefile]: I understand that - just showing the price of it.

Comment (mseri) [reply Makefile]: I agree with you by the way. In general I feel it's better to use build as the default, but I was convinced that the less the SPEC knows about this things the easier is to maintain them.

**Pull request merged**